Apps/WebClient/src/Server/Models/WalletConnectionModel.cs

```diff
@@ -0,0 +1,97 @@
+//-------------------------------------------------------------------------
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.WebClient.Models
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using HealthGateway.Common.Delegates;
+    using HealthGateway.Database.Constants;
+    using HealthGateway.Database.Models;
+
+    /// <summary>
+    /// Model that provides a user representation of an verifiable credential connection model.
+    /// </summary>
+    public class WalletConnectionModel
+    {
+        /// <summary>
+        /// Gets or sets the wallet connection id.
+        /// </summary>
+        public Guid WalletConnectionId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the user hdid.
+        /// </summary>
+        public string Hdid { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the issuer connection id.
+        /// </summary>
+        public Guid IssuerConnectionId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the connection date.
+        /// </summary>
+        public DateTime? ConnectedDate { get; set; }
+
+        /// <summary>
+        /// Gets or sets the disconnection date.
+        /// </summary>
+        public DateTime? DisconnectedDate { get; set; }
+
+        /// <summary>
+        /// Gets or sets the QR code data.
+        /// </summary>
+        public string? QrCode { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the connection state.
+        /// </summary>
+        public WalletConnectionStatus State { get; set; }
+
+        /// <summary>
+        /// Gets or sets the db record version.
+        /// </summary>
+        public uint Version { get; set; }
+
+        /// <summary>
+        /// Gets or sets the collection of credentials for this connection.
+        /// </summary>
+        public IEnumerable<WalletCredentialModel> Credentials { get; set; } = null!;
```

Comment:
Nonnullable. Where are we not going to use the JSON constructor for these and remove the setter?

Reply:
not sure why it didn't complain that it must be readonly here

Reply:
I think it is because it got initialized as null